apps/frontend/src/components/Spinner.tsx

```diff
@@ -0,0 +1,65 @@
+import CircularProgress from "@mui/material/CircularProgress";
```

Comment:
Is this component actually needed? In theory the loading state should be handled by whatever component is loading. And having the timeout here instead of on the request is also not how this should be done, as its not actually representative of the loading. And if an automatic timeout is being set, 40 seconds is way too long. 3-5 seconds is a more appropriate timeout

Reply:
The search/recommendation endpoint has taken over 30 seconds in the past to respond, so the default timeout on our axios requests is now 40 seconds until timeout. I know that this isn't technically ideal, but having a simple spinner makes the website look better and provides the user with some information other than a blank screen which may seem like a website failure to many users. 

The way it's set up now, it requires minimal effort to add it to our existing pages so I'd like to keep it like this if possible.